pkg/pgclient/client.go

```diff
@@ -33,84 +34,108 @@ type LockFunc = func(ctx context.Context, conn *pgx.Conn) error
 
 // Client sends Prometheus samples to TimescaleDB
 type Client struct {
-	Connection        pgxconn.PgxConn
-	QuerierConnection pgxconn.PgxConn
-	ingestor          *ingestor.DBIngestor
-	querier           querier.Querier
-	promqlEngine      *promql.Engine
-	healthCheck       health.HealthCheckerFn
-	queryable         promql.Queryable
-	ConnectionStr     string
-	metricCache       cache.MetricCache
-	labelsCache       cache.LabelsCache
-	seriesCache       cache.SeriesCache
-	closePool         bool
-	sigClose          chan struct{}
-	haService         *ha.Service
+	readerConn   pgxconn.PgxConn
+	ingestor     *ingestor.DBIngestor
+	querier      querier.Querier
+	promqlEngine *promql.Engine
+	healthCheck  health.HealthCheckerFn
+	queryable    promql.Queryable
+	metricCache  cache.MetricCache
+	labelsCache  cache.LabelsCache
+	seriesCache  cache.SeriesCache
+	connPools    []pgxconn.PgxConn
```

Comment:
Minor: Since there are two (reader and writer) + we have `readerConn` field already, it might be easier to just store `readerPool` and `writerPool`. Also, imho `PgxConn` (and using conn across the code) is missleading since this is actually a connection pool. I know this is not part of this change but just bringing it out there (so consider renaming `readerConn` to `readerPool`. We can rename`PgxConn` some other time.